lib/ansible/modules/cloud/azure/azure_rm_aks.py

```diff
@@ -96,6 +96,92 @@
         type: bool
         default: no
         version_added: 2.8
+    network_profile:
+        description:
+            - Profile of network configuration.
+        suboptions:
+            network_plugin:
+                description:
+                    - Network plugin used for building Kubernetes network.
+                    - This property cannot been changed.
+                choices:
+                    - azure
+                    - kubenet
+            network_policy:
+                description: Network policy used for building Kubernetes network.
+            pod_cidr:
+                description: A CIDR notation IP range from which to assign pod IPs when kubenet is used.
+            service_cidr:
+                description:
+                    - A CIDR notation IP range from which to assign service cluster IPs.
+                    - It must not overlap with any Subnet IP ranges.
+            dns_service_ip:
+                description:
+                    - An IP address assigned to the Kubernetes DNS service.
+                    - It must be within the Kubernetes service address range specified in serviceCidr.
+            docker_bridge_cidr:
+                description:
+                    - A CIDR notation IP range assigned to the Docker bridge network.
+                    - It must not overlap with any Subnet IP ranges or the Kubernetes service address range.
+        version_added: 2.8
+    aad_profile:
+        description:
+            - Profile of Azure Active Directory configuration.
+        suboptions:
+            client_app_id:
+                description: The client AAD application ID.
+            server_app_id:
+                description: The server AAD application ID.
+            server_app_secret:
+                description: The server AAD application secret.
+            tenant_id:
+                description:
+                    - The AAD tenant ID to use for authentication.
+                    - If not specified, will use the tenant of the deployment subscription.
+        version_added: 2.8
+    addon:
+        description:
+            - Profile of managed cluster add-on.
+            - Key can be C(http_application_routing), C(monitoring), C(virtual_node).
+            - Value must be a dict contains a bool variable C(enabled).
+        type: dict
+        suboptions:
+            http_application_routing:
+                description:
+                    - The HTTP application routing solution makes it easy to access applications that are deployed to your cluster.
+                type: dict
+                suboptions:
```

Comment:
do we need suboptions if it's jut one bool?

Reply:
I wanna keep consistency for all addons' parameters. Although http_application_routing has only one enabled parameter, other addons contain more than one.